### PR TITLE
Put detekt plugin config in a separate file to ease sharing

### LINKED ISSUE #13: Feature request: shared configuration
We have a shared project in the team, it would be nice to be able to share the project related config via version control. This would mean creating a separate file in `.idea`, instead of `workspace.xml` to store `<component name="DetektProjectConfiguration">` try to enable CheckStyle-idea on a project to see what I mean.

Additionally since the project would be shared a project-relative configuration support would be necessary. Currently only absolute paths work. we would like to use e.g. `config/detekt/detekt.yml`

### PR DESCRIPTION
**Pull request by bagage**
Fixes #13

:warning: there is no migration of previous configuration but I doubt it's critical, is it?

### CODE REVIEW

**Comment by arturbosch** [comment]
A warning in the changelog should be enough.

File: src/main/kotlin/io/gitlab/arturbosch/detekt/config/DetektConfigStorage.kt [hunk 1]
```diff
@@ -13,7 +13,7 @@ import com.intellij.util.xmlb.annotations.Tag
  */
 @State(
     name = "DetektProjectConfiguration",
-    storages = [(Storage(StoragePathMacros.WORKSPACE_FILE))]
+    storages = [Storage("detekt.kml")]
```
Comment (arturbosch) [inline src/main/kotlin/io/gitlab/arturbosch/detekt/config/DetektConfigStorage.kt]: kml is the right format?
And it is put by convention into the .idea folder?

Comment (bagage) [reply src/main/kotlin/io/gitlab/arturbosch/detekt/config/DetektConfigStorage.kt]: oops, I meant `.xml` obviously! Thanks for catching it.

Yes, path is relative to workspace by default, the file is created next to workspace.xml.

**Comment by bagage** [comment]
thanks for getting these PRs merged @arturbosch. Any chance to release a new plugin version soon? :)

**Pull request merged**